src/gui/components/CheckBox.zig

```diff
@@ -88,7 +90,7 @@ pub fn mainButtonReleased(self: *CheckBox, mousePosition: Vec2f) void {
 		self.pressed = false;
 		if(GuiComponent.contains(self.pos, self.size, mousePosition)) {
 			self.state = !self.state;
-			self.onAction(self.state);
+			self.onAction(self.ctx);
```

Comment:
This whole thing seems to be more fragile than it was before.
It should always give the callback the value that is visible to the user, instead of trusting the user to keep track of the state.